frontend/src/scenes/experiments/ExperimentView/components.tsx

```diff
@@ -101,17 +103,68 @@ export function ResultsQuery({
 }
 
 export function NoResultsEmptyState(): JSX.Element {
-    const { experimentResultsLoading, experimentResultCalculationError } = useValues(experimentLogic)
+    const { experimentResultsLoading, experimentResultCalculationError, experimentInsightType } =
+        useValues(experimentLogic)
+
+    function ChecklistItem({ failureReason, checked }: { failureReason: string; checked: boolean }): JSX.Element {
+        const failureReasonToText = {
+            'no-events': 'Events have been received',
+            'no-flag-info': 'Feature flag information is present on the events',
+            'no-control-variant': 'Events with the control variant received',
+            'no-test-variant': 'Events with at least one test variant received',
+        }
+
+        return (
+            <div className="flex items-center space-x-1">
+                {checked ? (
+                    <IconCheckbox className={`w-6 ${checked ? 'text-success' : ''}`} />
+                ) : (
+                    <IconSquare color="var(--muted)" fontSize="24" />
+                )}
+                <span>{failureReasonToText[failureReason]}</span>
+            </div>
+        )
+    }
 
     if (experimentResultsLoading) {
         return <></>
     }
 
+    if (experimentInsightType === InsightType.FUNNELS && experimentResultCalculationError) {
+        const checklistItems = []
+        for (const [failureReason, value] of Object.entries(JSON.parse(experimentResultCalculationError))) {
```

Comment:
not a big fan of this value being a JSON string. Can we parse it in the logic, add types for the shape of the result object, and then use that object here?

oh actually, thinking about this, the current code will break old experiments UI I think, because the API response is changing. Can you confirm that looks fine?

Reply:
1. yes, I want to move parsing into the logic once I implement this for Trends
2. will check 👍

Reply:
1. [ ] First
   1. [ ] second
    1. [ ] third